crates/hir-ty/src/method_resolution.rs

```diff
@@ -1167,10 +1167,13 @@ fn iterate_trait_method_candidates(
         // trait, but if we find out it doesn't, we'll skip the rest of the
         // iteration
         let mut known_implemented = false;
-        for &(_, item) in data.items.iter() {
+        'items: for &(_, item) in data.items.iter() {
+            if let AssocItemId::TypeAliasId(_) = item {
+                continue 'items;
+            }
```

Comment:
This is unnecessary, `is_valid_method_candidate` will immediately bail out already

Reply:
done!